ros2cli/ros2cli/cli.py

```diff
@@ -53,6 +64,17 @@ def main(*, script_name='ros2', argv=None, description=None, extension=None):
     # parse the command line arguments
     args = parser.parse_args(args=argv)
 
+    if not args.use_python_default_buffering:
+        # Make the output always line buffered.
+        # TextIoWrapper has a reconfigure() method, call that if available.
+        # https://docs.python.org/3/library/io.html#io.TextIOWrapper.reconfigure
+        try:
+            sys.stdout.reconfigure(line_buffering=True)
+        except AttributeError:
+            # if stdout is not a TextIoWrapper instance, or we're using python older than 3.7,
+            # force line buffering by patching print
+            builtins.print = functools.partial(print, flush=True)
```

Comment:
@ivanpauno FYI if the decorator monitors the `file` keyword argument of `print`, you can limit its action to `sys.stdout`.

Reply:
true, I didn't feel that was really needed though.
do you think we need a patch for this?

Reply:
Nope, it's unlikely anyone will ever hit it before Python 3.6 support is gone.